hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/error/TestHoodieBackedErrorTable.java

```diff
@@ -0,0 +1,100 @@
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hudi.client.SparkRDDWriteClient;
+import org.apache.hudi.client.WriteStatus;
+import org.apache.hudi.common.config.HoodieErrorTableConfig;
+import org.apache.hudi.common.model.HoodieKey;
+import org.apache.hudi.common.model.HoodieRecord;
+import org.apache.hudi.common.model.OverwriteWithLatestAvroPayload;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.testutils.HoodieTestDataGenerator;
+import org.apache.hudi.config.HoodieMemoryConfig;
+import org.apache.hudi.config.HoodieWriteConfig;
+import org.apache.hudi.testutils.HoodieClientTestBase;
+import org.apache.spark.api.java.JavaRDD;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.apache.hudi.testutils.Assertions.assertHasWriteErrors;
+import static org.apache.hudi.testutils.Assertions.assertNoWriteErrors;
+import static org.apache.hudi.common.testutils.HoodieTestDataGenerator.TRIP_SCHEMA;
+import static org.apache.hudi.common.testutils.Transformations.recordsToHoodieKeys;
+import static org.apache.hudi.common.util.ParquetUtils.readRowKeysFromParquet;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestHoodieBackedErrorTable extends HoodieClientTestBase {
+
+  @Test
+  public void testInsertError() {
+    SparkRDDWriteClient writeClient = getHoodieWriteClient(getConfig());
+    String commitTime = writeClient.startCommit();
+    List<HoodieRecord> hoodieRecords = dataGen.generateInserts(commitTime, 100);
+
+    JavaRDD<HoodieRecord> recordsRDD = jsc.parallelize(hoodieRecords, 1);
+    JavaRDD<WriteStatus> writeStatusRDD = writeClient.insertError(recordsRDD, commitTime);
+    List<WriteStatus> statuses = writeStatusRDD.collect();
+    assertNoWriteErrors(statuses);
+  }
+
+  @Test
+  public void testHasErrorRecordInsert() {
+    HoodieWriteConfig config1 = getConfigBuilder(TRIP_SCHEMA)
+            .withMemoryConfig(HoodieMemoryConfig.newBuilder().withWriteStatusFailureFraction(1).build())
+            .withErrorTableConfig(HoodieErrorTableConfig.newBuilder().enable(true).build()).build();
+    SparkRDDWriteClient writeClient1 = getHoodieWriteClient(config1);
+    final String testPartitionPath = "2020/03/17";
+    dataGen = new HoodieTestDataGenerator(new String[] {testPartitionPath});
+
+    String commitTime1 = writeClient1.startCommit();
+    List<HoodieRecord> hoodieRecords1 =
+            dataGen.generateInsertsStream(commitTime1, 50, false, TRIP_SCHEMA).collect(Collectors.toList());
+    List<HoodieKey> insertKeys1 = recordsToHoodieKeys(hoodieRecords1);
+    List<WriteStatus> writeStatuses = insertAndCheck(writeClient1, insertKeys1, commitTime1);
+
+    assertHasWriteErrors(writeStatuses);
+    assertEquals(50, writeStatuses.get(0).getFailedRecords().size(), "shoule contain 50 records");
```

Comment:
shoule contain 50 records  -> should contain 50 records

Reply:
My mistake